From e5ed82a1719ebabda8d2ae7b02ab59dc1c819b4b Mon Sep 17 00:00:00 2001
From: John Stebbins <stebbins@jetheaddev.com>
Date: Thu, 22 Dec 2016 09:23:30 -0800
Subject: [PATCH] matroskadec: fix SRT subtitle duration

The codec id for SRT was changed and conditionals were not updated.
---
 libavformat/matroskadec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavformat/matroskadec.c b/libavformat/matroskadec.c
index 8016730..a3954b0 100644
--- a/libavformat/matroskadec.c
+++ b/libavformat/matroskadec.c
@@ -2439,11 +2439,11 @@ static int matroska_parse_frame(MatroskaDemuxContext *matroska,
     else
         pkt->pts = timecode;
     pkt->pos = pos;
-    if (track->type != MATROSKA_TRACK_TYPE_SUBTITLE || st->codecpar->codec_id == AV_CODEC_ID_TEXT)
+    if (track->type != MATROSKA_TRACK_TYPE_SUBTITLE || st->codecpar->codec_id == AV_CODEC_ID_SRT)
         pkt->duration = duration;
 #if FF_API_CONVERGENCE_DURATION
 FF_DISABLE_DEPRECATION_WARNINGS
-    if (st->codecpar->codec_id == AV_CODEC_ID_TEXT)
+    if (st->codecpar->codec_id == AV_CODEC_ID_SRT)
         pkt->convergence_duration = duration;
 FF_ENABLE_DEPRECATION_WARNINGS
 #endif
-- 
2.7.4

